apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1283,37 +1283,36 @@ new_onchain_tx_for_signing(Type, Opts, D) ->
     end.
 
 new_onchain_tx_for_signing_(Type, Opts, D) ->
-    FeeSpecified = maps:is_key(fee, Opts),
     Defaults = tx_defaults(Type, Opts, D),
     Opts1 = maps:merge(Defaults, Opts),
-    {ok, Tx, Updates} = new_onchain_tx(Type, Opts1, D),
-    CurHeight = cur_height(),
-    case {aetx:min_fee(Tx, CurHeight), aetx:fee(Tx)} of
+    CurrHeight = curr_height(),
+    {ok, Tx, Updates} = new_onchain_tx(Type, Opts1, D, CurrHeight),
+    case {aetx:min_fee(Tx, CurrHeight), aetx:fee(Tx)} of
         {MinFee, Fee} when MinFee =< Fee ->
             {ok, Tx, Updates};
-        {MinFee, Fee} when FeeSpecified ->
-            lager:debug("Specified fee (~p) is too low for ~p (Min: ~p)",
-                        [Fee, Type, MinFee]),
-            error(too_low_fee);
         {MinFee, Fee} ->
-            lager:debug("Must adjust fee (Fee: ~p, Min: ~p)", [Fee, MinFee]),
-            %% adjust fee to be at least the minimum for this tx
-            new_onchain_tx(Type, Opts1#{fee => trunc(MinFee * 1.1)}, D)
+            lager:debug("Fee (~p) is too low for ~p (Min: ~p)",
+                        [Fee, Type, MinFee]),
+            error(too_low_fee)
     end.
 
 -spec new_onchain_tx(channel_create_tx
                    | channel_close_mutual_tx
                    | channel_deposit_tx
-                   | channel_withdraw_tx, map(), #data{}) ->
+                   | channel_withdraw_tx
+                   | channel_close_solo_tx
+                   | channel_slash_tx
+                   | channel_settle_tx, map(), #data{}, aec_blocks:height()) ->
     {ok, aetx:tx(), [aesc_offchain_update:update()]}.
 new_onchain_tx(channel_close_mutual_tx, #{ acct := From } = Opts,
-               #data{opts = DOpts, on_chain_id = Chan, state = State}) ->
+               #data{opts = DOpts, on_chain_id = Chan, state = State}, _) ->
     #{initiator := Initiator,
       responder := Responder} = DOpts,
     ChanId = aeser_id:create(channel, Chan),
     FromId = aeser_id:create(account, From),
     {ok, IAmt} = aesc_offchain_state:balance(Initiator, State),
     {ok, RAmt} = aesc_offchain_state:balance(Responder, State),
+    % fee is preset. This is important for participant's amounts calculation
```

Comment:
Use `%%` instead of `%` when comment is expected to line up with existing indentation. Otherwise, Emacs will send it to far-right if asked to fix indentation.